### PR TITLE
Fixing bug with 'cordova prepare'

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: tools/after-platform-add-android.js [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
         data = data.replace(/<\/activity>/, "</activity>-->");
         
-        // Change min sdk version
-        data = data.replace(/android\:minSdkVersion\=\"10\"/, 'android:minSdkVersion="19"');
-        
-        // Change target api
-        data = data.replace(/android\:targetSdkVersion\=\"22\"/, 'android:targetSdkVersion="' + targetAndroidApi + '"');
-        
+        // Change min SDK version.
+        data = data.replace(/android\:minSdkVersion\=\"14\"/, 'android:minSdkVersion="19"');
```
Comment (bhariharan) [inline tools/after-platform-add-android.js]: This is a different bugfix. I took the opportunity to cleanup some incorrect steps we had.

**Comment by bhariharan** [comment]
Took me a while to get to the bottom of this, but here's the summary of my findings.

* Our plugin requires that we install it last, so that we can run our post-install script. This script basically replaces the default ```MainActivity``` entry in the manifest with our own ```SalesforceDroidGapActivity```. However, the entries in the manifest are added in the order the plugins are added. So, naturally, ```PushHandlerActivity``` comes along first and ours is at the bottom. In a typical ```Android``` environment, this order doesn't matter.
* As part of our creation process, we run ```cordova prepare``` at the end, and this is where the nightmare begins.
* ```cordova prepare```, in its effort to be really smart (NOT), reads the first ```Activity``` it can come across in the manifest, and adds a tag ```android:launchMode="singleTop"``` to it. The correct behavior here on Cordova's part should be to add that to the launch activity, not the first one on the list.
* Now, when we manually run the second ```cordova prepare```, it doesn't know that there's already an entry for ```PushHandlerActivity```, because instead of looking for the activity itself, it searches for an exact match to the full line. However, this won't match, because it incorrectly added ```android:launchMode="singleTop"``` to it, in the previous step. Hence, it ends up adding a second entry for ```PushHandlerActivity```.

Here's how I've solved this for now.
* I have added a post-prepare hook to remove the entry for ```singleTop```. This means that when the first ```cordova prepare``` is run, we will remove ```singleTop```, and when the second ```cordova prepare``` is run, it won't add the duplicate entry.
* Enter ```Cordova``` bug number 2 - it doesn't run the post-prepare hook every time ```cordova prepare``` is run, it only runs it the first time or if there are changes in the ```www``` folder. But it does the ```prepare``` config stuff that plugins require. So, if the user did run a ```cordova prepare``` a third time with no changes, we'd be back to square 1. I'll file issues with ```Cordova``` for these, but for now, we should be good.

**Comment by wmathurin** [comment]
👍 
And that's why we love Cordova ;-)

**Comment by BrettMN** [comment]
@bhariharan Thanks for the the work on this and the thorough explanation!

File: tools/after-platform-add-android.js (hunk 1 above)
Comment (kchitalia) [inline tools/after-platform-add-android.js]: So no need to replace the target api version anymore?

Comment (bhariharan) [inline tools/after-platform-add-android.js]: No.

**Pull request merged**